BitwardenShared/UI/Vault/Vault/AutofillList/VaultAutofillListProcessor.swift

```diff
@@ -152,6 +163,76 @@ class VaultAutofillListProcessor: StateProcessor<
         }
     }
 
+    @available(iOSApplicationExtension 17.0, *)
+    private func initFido2State() {
+        guard let fido2appExtensionDelegate = appExtensionDelegate as? Fido2AppExtensionDelegate else {
+            return
+        }
+
+        services.fido2UserInterfaceHelper.setupDelegate(fido2UserVerificationMediatorDelegate: self)
```

Comment:
🤔 Is the delegate that is passed in here only used for presenting alerts? Or does/will it have other uses? Instead of setting a delegate, I wonder if we could pass a closure that would be used for presenting alerts, such that you could pass `coordinator.showAlert`? I'm not sure how many methods that would change, but it was a thought I had.   Or maybe there's another way we could give it a way to present an alert (like have app processor set the app coordinator as the delegate).

Reply:
It shows alerts and it's also used to perform `setupPin` navigation for user verification which is the last part of the flow if the other mechanisms aren't available on user verification preference `required`.